module_utils/network_lsr/argument_validator.py

```diff
@@ -413,6 +414,73 @@ def _validate_post(self, value, name, result):
 
         return result
 
+class ArgValidatorIPRule(ArgValidatorDict):
+    def __init__(self, name, family=None, required=False, default_value=None):
+        ArgValidatorDict.__init__(
+            self,
+            name,
+            required,
+            nested=[
+                ArgValidatorIP(
+                    "from", family=family, required=False, plain_address=False
+                ),
+                ArgValidatorNum("from_prefix", default_value=32, val_min=0, val_max=32),
+                ArgValidatorIP(
+                    "to", family=family, required=False, plain_address=False
+                ),
+                ArgValidatorNum("to_prefix", default_value=32, val_min=0, val_max=32),
+                ArgValidatorStr(
+                    "iif", default_value=None, enum_values=listdir('/sys/class/net/')
+                ),
+                ArgValidatorStr(
+                    "oif", default_value=None, enum_values=listdir('/sys/class/net/')
+                ),
+                ArgValidatorNum(
+                    "tos", default_value=None, val_min=0, val_max=255
+                ),
+                ArgValidatorNum(
+                    "dsfield", default_value=None, val_min=0, val_max=255
+                ),
+                ArgValidatorNum(
+                    "fwmark", default_value=None, val_min=0, val_max=65535
+                ),
+                # Not implemented
+                # ArgValidatorRange(
+                #     "uidrange", default_value=None, val_min=0, val_max=65535
+                # ),
+                ArgValidatorNum(
+                    "ipproto", default_value=None, val_min=0, val_max=255
+                ),
+                # ArgValidatorRange(
+                #     "sport", default_value=None, val_min=0, val_max=65535
+                # ),
+                # ArgValidatorDport(
+                #     "dport", default_value=None, val_min=0, val_max=65535
+                # ),
+                ArgValidatorNum(
+                    "priority", default_value=None, val_min=0, val_max=32767
```

Comment:
priority ranges up to `2^32-1`.

Also, a priority of zero means that kernel will automatically assign a priority. I think that the rule should not allow for that, because the priority is one of the most important parameters and specifying multiple rules with the same priority seems error prone. I think the user should specify this parameter. Or otherwise, the role should always generate it (e.g. by taking the index of the rule in the playbook).

Reply:
There is a bug in kernel (and all it's patches, 3.10-xxx) in RH/Centos 7.x, which allows adding multiple rules with the same priority. That has very bad impact on this role :/ They promised the fix should make it into 7.8 next year.

So talking about "importance" here ... I will fix the range.

Reply:
Right (https://bugzilla.redhat.com/show_bug.cgi?id=1700691).

But omitting the priority doesn't solve that, does it?

If you issue `ip rule delete ...` without specifying a priority, then kernel will delete the first rule that matches the remaining parameters. I don't think that is desirable and omitting the priority only makes it worse.

Reply:
No, omitting priority is not an option. In such case, the role should have a deterministic system of numbering rules for the user, which would be probably hard to manage.

So we should probably just update README warning users about this potential issue.